scipy/sparse/compressed.py

```diff
@@ -846,8 +846,10 @@ def _insert_many(self, i, j, x):
         # update attributes
         self.indices = np.concatenate(indices_parts)
         self.data = np.concatenate(data_parts)
-        nnzs = np.asarray(np.ediff1d(self.indptr, to_begin=0), dtype=idx_dtype)
-        nnzs[1:][ui] += new_nnzs
+        nnzs = np.empty(self.indptr.shape, dtype=idx_dtype)
+        nnzs[0] = idx_dtype(0)
+        nnzs[1:] = np.diff(self.indptr).astype(idx_dtype)
```

Comment:
Pass `copy=False` to `astype` to avoid extra copying in some cases.

Reply:
Though now that I consider it, I'm not sure the `astype` call is necessary at all. Assignment to an array slice preserves the original array's `dtype`.

Reply:
I've left it as that in a defensive manner because I wasn't sure what the dtypes are that is being processed here. In other words if `nnz` is `int32` and `self.indptr` was for some reason `int64` it would be keeping the int32 behavior instead of upcasting but maybe too safe?

Reply:
It'll preserve the dtype of `nnzs` (and thus `idx_dtype`), regardless of the dtype of `self.indptr`.

So in your example, no upcasting would occur (as intended).